cubecrit/controllers/search.py

```diff
@@ -0,0 +1,44 @@
+"""Routes for search pages."""
+from flask import Blueprint, abort, current_app, render_template, request
+
+from ..controllers.puzzles import validate_page_number
+from ..models.puzzle import Puzzle, PuzzleType
+
+search = Blueprint(
+    "search", __name__, template_folder="templates", url_prefix="/search"
+)
+
+
+@search.route("/", methods=["GET"])
+def get_search_route() -> str:
+    """Get a search for puzzles.
+
+    Raises:
+    - a 404 error if the page number is invalid.
+
+    Returns an HTML page.
+    """
+    puzzle_type = request.args.get("puzzle_type")
+    page = validate_page_number(request.args.get("page"))
+    print(type(puzzle_type))
+    if puzzle_type is not None and puzzle_type.strip() == "":
+        puzzle_type = None
+    print(type(puzzle_type))
+    query = request.args.get("query")
+    if query is not None and query.strip() == "":
```

Comment:
We should refactor the logic for validating the `query and `puzzle_type` into a helper function, since we'll probably re-use it for other filters in the future.

Reply:
ignore, we'll do later with #25